cmd/node-termination-handler.go

```diff
@@ -150,14 +151,15 @@ func main() {
 			time.Sleep(8 * time.Second)
 			stopCh <- struct{}{}
 		}()
+
 		//will retry 4 times with an interval of 2 seconds.
-		err = wait.PollImmediateUntil(2*time.Second, func() (done bool, err error) {
+		err = wait.PollUntilContextCancel(wait.ContextForChannel(stopCh), 2*time.Second, true, func(context.Context) (done bool, err error) {
 			err = handleRebootUncordon(nthConfig.NodeName, interruptionEventStore, *node)
 			if err != nil {
 				log.Warn().Err(err).Msgf("Unable to complete the uncordon after reboot workflow on startup, retrying")
 			}
 			return false, nil
-		}, stopCh)
+		})
```

Comment:
... continuation of previous suggestion

```suggestion
		})
		cancelPollCtx()
```